plutus-tx/src/PlutusTx/Code.hs

```diff
@@ -57,34 +57,51 @@ type CompiledCode = CompiledCodeIn PLC.DefaultUni PLC.DefaultFun
 
 -- | Apply a compiled function to a compiled argument. Will fail if the versions don't match.
 applyCode
-    :: (PLC.Closed uni, uni `PLC.Everywhere` Flat, Flat fun)
-    => CompiledCodeIn uni fun (a -> b) -> CompiledCodeIn uni fun a -> Maybe (CompiledCodeIn uni fun b)
+    :: (PLC.Closed uni
+        , uni `PLC.Everywhere` Flat
+        , Flat fun
+        , Pretty fun
+        , PLC.Everywhere uni PrettyConst
+        , PrettyBy RenderContext (PLC.SomeTypeIn uni))
+    => CompiledCodeIn uni fun (a -> b)
+    -> CompiledCodeIn uni fun a
+    -> Either String (CompiledCodeIn uni fun b)
```

Comment:
I can add an error type for this too but I don't think it's beneficial in this case.

Reply:
Yeah, seems fine.